### PR TITLE
`r/azurerm_disk_encryption_set`: add ID parsing function

### PR DESCRIPTION
**Pull request by ArcturusZhang**

### CODE REVIEW

File: azurerm/internal/services/compute/resource_arm_managed_disk.go [hunk 1]
```diff
@@ -144,7 +145,7 @@ func resourceArmManagedDisk() *schema.Resource {
 				// TODO: make this case-sensitive once this bug in the Azure API has been fixed:
 				//       https://github.com/Azure/azure-rest-api-specs/issues/8132
 				DiffSuppressFunc: suppress.CaseDifference,
-				ValidateFunc:     azure.ValidateResourceID,
+				ValidateFunc:     validate.DiskEncryptionSetID,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/resource_arm_managed_disk.go]: unfortunately we can't use this here until the API returns this in the correct case

Comment (ArcturusZhang) [reply azurerm/internal/services/compute/resource_arm_managed_disk.go]: Done

**Comment by ArcturusZhang** [comment]
Hi @tombuildsstuff I have revert the validation in `azurerm_managed_disk`.

File: azurerm/internal/services/compute/resource_arm_managed_disk.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/internal/services/compute/resource_arm_managed_disk.go]: @ArcturusZhang I think you may have forgotten to push that commit? :)

Comment (ArcturusZhang) [reply azurerm/internal/services/compute/resource_arm_managed_disk.go]: 😂how silly of me

**Comment by ArcturusZhang** [comment]
Hi @tombuildsstuff well, this time I have pushed the changes. 😂

**Pull request merged**